### PR TITLE
Lint/test the UI on the CI config

### PR DESCRIPTION
**Pull request by alahwa**
I actually overestimated what CircleCI was capable of. Husky will continue to manage our precommit behavior (lint on precommit etc.) and CI will run tests and verify that code is properly linted.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -5,4 +5,24 @@ jobs:
       - image: node
     steps:
       - checkout
-      - run: echo "A first hello"
+      - restore_cache:
+          key: node-modules-{{ .Branch }}--{{ checksum "ui/package.json" }}
```
Comment (melissachang) [inline .circleci/config.yml]: nit Can you change `--` to `-`, latter seems to be standard

Same on line 16

File: .circleci/config.yml [hunk 2]
```diff
@@ -5,4 +5,24 @@ jobs:
...
+          name: nvm install node and UI dependencies
+          command: |
+            cd ui
+            npm install
+      - save_cache:
+          key: node-modules-{{ .Branch }}--{{ checksum "ui/package.json" }}
+          paths:
+            - "ui/node_modules"
+      - run:
+          name: Running the linter
```
Comment (melissachang) [inline .circleci/config.yml]: nit I prefer job manager style, "Run linter" (Running -> Run)

File: .circleci/config.yml [hunk 3]
```diff
@@ -5,4 +5,24 @@ jobs:
...
+          command: |
+              cd ui
+              npm run lint -- --list-different
+      - run:
+          name: Running ui unit tests
```
Comment (melissachang) [inline .circleci/config.yml]: Run UI unit tests
(ui -> UI)

**Comment by alahwa** [comment]
All done. Also for the record, caching doesn't work locally so verified that a cache was created in [7](https://circleci.com/gh/DataBiosphere/data-explorer/7) and manually reran the CI to ensure the cache was retrieved in [8](https://circleci.com/gh/DataBiosphere/data-explorer/7)

**Pull request merged**